witness/ethereum/contracts/CompactRangeWitness.sol

```diff
@@ -0,0 +1,251 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.0;
+
+import {WitnessUtils} from "./WitnessUtils.sol";
+
+/// @author sinasabet@google.com
+/// @title Witness tracking external Merkle Tree via compact ranges.
+contract CompactRangeWitness is WitnessUtils {
+    /// @dev struct to encapsulate data for a specific tree.
+    struct TreeData {
+        bytes32[] compactRange;
+        bytes32 signature;
+        uint256 size;
+        uint256 timestamp;
+    }
+
+    /// @dev emits information related to an update method call.
+    event Updated(
+        bytes32 treeId,
+        bytes32 root,
+        uint256 treeSize,
+        uint256 timestamp,
+        bytes32[] appendedCompactRange,
+        bytes32 signature
+    );
+
+    address public owner;
+    /// @dev mapping from treeId to that tree's data.
+    mapping(bytes32 => TreeData) public treeDatas;
+
+    constructor() {
+        owner = msg.sender;
+    }
```

Comment:
I guess this `Ownable` logic can be moved to another contract, it's repeated in both witness